modal/_utils/task_command_router_client.py

```diff
@@ -458,25 +463,28 @@ async def sleep_and_update_delay_and_num_retries_remaining(e: Exception):
                         file_descriptor=file_descriptor,
                     )
 
-                    # Scope auth retry strictly to the initial send (where headers/auth are sent).
+                    # Auth retry is scoped to a single refresh per streaming attempt. While auth metadata is
+                    # sent on request start, UNAUTHENTICATED may sometimes surface during iteration,
+                    # so we handle it at both send and receive boundaries.
                     try:
                         await s.send_message(req, end=True)
+                        async for item in s:
+                            # We successfully authenticated after a JWT refresh, reset the auth retry flag.
+                            if did_auth_retry:
+                                did_auth_retry = False
```

Comment:
Is this the same as this? 🧐

```suggestion
                            did_auth_retry = False
```